### PR TITLE
[AIRADSW-852] Fixing regression with slice-squeeze rewrite for scalar inputs

### PR DESCRIPTION
**Pull request by urpetkov-amd**
## Motivation

This [PR #5004](https://github.com/ROCm/AMDMIGraphX/pull/5004) caused a regression.

Fix a `simplify_reshapes` compilation failure for graphs containing a
`slice -> squeeze -> pointwise` sequence with a scalar companion input.

The `find_slice_squeeze` rewrite moves the `squeeze` after the pointwise
operation and restores the removed dimension on the operation's other inputs.
It previously handled every companion input with `unsqueeze`. For scalar
inputs, this did not produce dimensions compatible with the unsqueezed slice,
so pointwise shape validation failed with:

```text
same_dims: mul: Dimensions do not match
```

This prevented affected ONNX models from compiling.

## Technical Details

- Detect scalar companion inputs in the `find_slice_squeeze` rewrite.
- Restore scalar inputs with `multibroadcast` using the sliced input's output
  dimensions.
- Continue using `unsqueeze` for non-scalar companion inputs.
- Add the `slice_squeeze_binary_scalar` regression test to cover a
  `slice -> squeeze -> mul(scalar)` graph and verify its final canonical form.

This corrects the scalar edge case in the slice/squeeze matcher added by
[PR #5004](https://github.com/ROCm/AMDMIGraphX/pull/5004).

## Testing

- Compiled `cpe-v2-fp16-384x672.onnx` successfully with
  `migraphx-driver compile`.
- Ran `test_simplify_reshapes_test.exe`: all 205 tests passed.

## Changelog Category

Add a `CHANGELOG.md` entry for any option other than `Not Applicable`
- - [ ] Added: New functionality.
- - [ ] Changed: Changes to existing functionality.
- - [ ] Removed: Functionality or support that has been removed. (Compared to a previous release)
- - [ ] Optimized: Component performance that has been optimized or improved.
- - [x] Resolved Issues: Known issues from a previous version that have been resolved.
- - [ ] Not Applicable: This PR is not to be included in the changelog.

Follow the [LLVM AI Tool Use Policy](https://llvm.org/docs/AIToolPolicy.html) for contributions using AI.

### CODE REVIEW

**Review by pfultz2 — APPROVED** [review]

File: test/simplify_reshapes_test.cpp [hunk 1]
```diff
@@ -5334,6 +5334,33 @@ TEST_CASE(slice_squeeze_binary_two_inputs)
...
+    }
+    run_pass(m1);
+    migraphx::module m2;
+    {
+        auto input   = m2.add_parameter("input", s);
+        auto scalar  = m2.add_literal(0.5f);
+        auto squeeze = m2.add_instruction(migraphx::make_op("squeeze", {{"axes", {0}}}), input);
+        auto slice   = m2.add_instruction(
+            migraphx::make_op("slice", {{"axes", {0}}, {"starts", {0}}, {"ends", {1}}}), squeeze);
+        auto mul = m2.add_instruction(migraphx::make_op("mul"), slice, scalar);
```
Comment (CharlieL7) [inline test/simplify_reshapes_test.cpp]: The order of the squeeze here is different from the other cases. It is going `squeeze -> slice -> mul` rather than `slice -> mul -> squeeze`. I think there is another matcher (find_op_shape_transform_op?) that pulls up the `squeeze` instruction. Could you add a comment to this test case that it's hitting `find_slice_squeeze` and the other matcher?

**Review by CharlieL7 — CHANGES_REQUESTED** [review]

File: test/simplify_reshapes_test.cpp (hunk 1 above)
Comment (urpetkov-amd) [reply test/simplify_reshapes_test.cpp]: I changed that and added a comment.

Matcher tracing showed the original test hit only `find_slice_shape_transforms`, not `find_slice_squeeze`. The revised test keeps a second input use, preventing preemption and directly testing `find_slice_squeeze`.

**Review by TedThemistokleous — APPROVED** [review]

**Comment by urpetkov-amd** [comment]
I resolved recent conflicts.

**Comment by urpetkov-amd** [comment]
I updated a branch and I started working on fixing this HIP Clang Release issues that happened in CI in order merge it soon.

**Comment by urpetkov-amd** [comment]
@causten 

Issues have been resolved. All jobs have passed.

**Review by CharlieL7 — APPROVED** [review]
